core/src/main/java/com/linecorp/armeria/common/AbstractContextAwareScheduledExecutorService.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+abstract class AbstractContextAwareScheduledExecutorService extends AbstractContextAwareExecutorService
+        implements ScheduledExecutorService {
+    final ScheduledExecutorService executor;
```

Comment:
Could remove this field and add a method like this:

```java
private ScheduledExecutorService unwrap() {
    return (ScheduledExecutorService) executor;
}
```

.. because there's no need to keep the same reference in two fields.

Reply:
The existing implementation did it this way; I don't like the unsafe cast, but I think I can see how to do it with a generic.